web/packages/teleport/src/Sessions/SessionList/SessionJoinBtn.tsx

```diff
@@ -82,28 +81,91 @@ function JoinMenu({ children }: { children: React.ReactNode }) {
       <Menu
         anchorOrigin={{
           vertical: 'bottom',
-          horizontal: 'center',
+          horizontal: 'right',
         }}
         transformOrigin={{
           vertical: 'top',
-          horizontal: 'center',
+          horizontal: 'right',
         }}
         anchorEl={anchorEl}
         open={Boolean(anchorEl)}
         onClose={handleClose}
       >
-        <Text
-          px="2"
-          fontSize="11px"
-          css={`
-            color: ${props => props.theme.colors.text.primary};
-            background: ${props => props.theme.colors.spotBackground[2]};
-          `}
-        >
-          Join as...
-        </Text>
         {children}
       </Menu>
     </Box>
   );
 }
+
+function JoinMenuItem({
+  title,
+  description,
+  hasAccess,
+  participantMode,
+  url,
+}: {
+  title: string;
+  description: string;
+  hasAccess: boolean;
+  participantMode: ParticipantMode;
+  url: string;
+}) {
+  if (hasAccess) {
+    return (
+      <MenuItem
+        as="a"
+        href={url}
+        target="_blank"
+        css={`
+          text-decoration: none;
+          padding: 8px 12px;
+          color: ${({ theme }) => theme.colors.text.primary};
+          user-select: none;
+          border-bottom: 1px solid
+            ${({ theme }) => theme.colors.spotBackground[0]};
+        `}
+      >
+        <Box height="fit-content" width="264px">
+          <Text typography="h6">{title}</Text>
+          <Text color="text.secondary">{description}</Text>
+        </Box>
+      </MenuItem>
+    );
+  }
+  return (
+    <MenuItem
+      css={`
+        text-decoration: none;
+        padding: 8px 12px;
+        color: ${({ theme }) => theme.colors.text.disabled};
+        user-select: none;
+        cursor: auto;
+        border-bottom: 1px solid
+          ${({ theme }) => theme.colors.spotBackground[0]};
+        &:hover {
+          background-color: ${({ theme }) => theme.colors.levels.elevated};
+          color: ${({ theme }) => theme.colors.text.disabled};
+        }
+      `}
+    >
+      <Box height="fit-content" width="264px">
+        <Text typography="h6">{title}</Text>
+        <Text>{description}</Text>
+        <Box color="text.primary" px={1} mt={1}>
+          <Text fontSize="10px" color="text.secondary">
+            <Warning color="error.main" mr={2} />
+            You do not have permission to join as a
+            {participantMode === 'observer' ? 'n ' : ' '}
```

Comment:
This is clever. Another way to maybe do this is have an "warning map" or something like
```js
const WARNING_TEXT = {
    observer: "You do not have permission to join as an observer",
    moderator: "You do not have permission to join as a moderator",
    peer: "You do not have permission to join as a peer"
}

then

{WARNING_TEXT[participantMode]}

```

not necessary but just offering another option. You pick.